sdk-python/create_bulk_messages.py

```diff
@@ -32,57 +35,70 @@ def get_next_tag(headers):
         return CommonConstants.TAGS_FIELD
 
 
-def schedule_calls(csv_file_path):
+def create_bulk_messages(headers, message_data, file_path):
     """
-    :param csv_file_path: Path to the csv file where messages to be sent are present
-    :type csv_file_path: String
+    :param headers: Headers of the data
+    :type headers: List
+    :param message_data: Message data eg: [{phone_number:8929292929,send_on:"",tag1:"tag_number1",template:23,language:"hi"}]
+    :type message_data: List of dict
+    :param file_path: Path to the folder where the file with created/pending messages are to be kept
+    :type file_path: String
     :return: Creates another csv_files in the folder where the csv file was present
             with created and not created messages
     """
-    ad_manager = ADUtils(organization, username, password)
-    created_messages = None
-    limit = CommonConstants.AD_MESSAGE_SCHEDULE_LIMIT
     message_request_id = str(uuid.uuid1())
-    file_path = os.path.dirname(csv_file_path)
+    ad_manager = APIUtils()
     try:
         """
-        Step 1: Get Messages from CSV file
-        """
-        headers, message_data = CSVUtils.parse_csv(csv_file_path)
-
-        """
-        Step 2:  Mark the messages sent in this particular request with a unique request
-                 is as the next tag, use this later to check the status of the messages.
+        Step 1:  Mark the messages sent in this particular request with a unique request
+                 is as the next tag,can use this later to check the status of the messages.
         """
         tag = get_next_tag(headers)
         for item in message_data:
             item.update({tag: message_request_id})
         """
-        Step 3: Schedule messages in chunks
+        Step 2: Schedule messages and  Create file for created messages and pending messages if any 
         """
-        ad_manager.bulk_create_in_chunks(message_data, limit)
-        scheduling_completed = True
+        kwargs = {'limit': args.limit, 'use_custom_format': args.transform_with_template}
+        created_messages = ad_manager.create_bulk_in_chunks(awaazde_api.messages, message_data, **kwargs)
+        CSVUtils.write_csv(created_messages, file_path, file_name="created")
 
     except Exception as e:
-        scheduling_completed = False
         logging.error("Error occurred trying to schedule calls : {}".format(e))
```

Comment:
Is this entire manual error handling (and even creation of a tag to track messages) necessary? Up above you are getting the messages back chunk by chunk. In the response itself you are getting back the success/error status of each message. Why not just simply write out everything into a single file and let the user deal with it? This is EXACTLY the same as what user gets by downloading the import summary from notification. They should be able to deal with it just fine that way.

If you do that I think you can get rid of this entire exception. In the very rare case that there is some unexpected failure in the middle of the chunked, they user can manually diff the import summary with the original file, or use the import_and_match command. But again, I just don't see that this is going to happen enough for it to be useful